com.github.scrivanolabs.scrivano.yml

```diff
@@ -0,0 +1,32 @@
+app-id: com.github.scrivanolabs.scrivano
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: Scrivano
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-config
+  - --filesystem=xdg-desktop
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-download
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-public-share
```

Comment:
Not sure this is wise either.

Reply:
Could you please explain why this is not wise? Are you referring to the xdg-public-share permission? What do you think would be the most appropriate set of permissions for a notetaking application?

Reply:
you should explain why you want to read-write all these location including `xdg-public-share`.

Reply:
Scrivano is a notetaking application and allows users to save their notes in a folder of their choice on their machine (personally, as a user I would like to have the freedom to save my notes anywhere, at least in my home directory, but I understand that Flatpak users might not want this).

To do this, Scrivano uses a custom file dialog built-in within the app (this choice was made of several reasons, primarily because of consistency across operating systems and easily display thumbnails for .scrvn files and also keeping in mind that in future an integrated sidebar will likely be displayed inside the application to allow users to easily switch between their notes). I've read on the Flatpak website that an app should first consider using portals to let users use a a native file dialog to access this, but this wouldn't be possible with Scrivano (also because the app requires the ability to auto-save files based on a time interval without the user directly intervening).

So I think granting access to xdg-documents is quite important. Scrivano also needs the ability to import/export PDF files and pictures thus it needs to access xdg-downloads and xdg-pictures folders. I think one could argue that only read-only access should be given to these folders, however users often want to save their notes in these folders and I think it would be quite limiting to not be able to do so. For example, a scenario I run into often is this: I have a PDF I would like to annotate so I import it in Scrivano and then save it next to the file until I finish annotating it and then eventually export it as a PDF in a folder of my choice.
For the xdg-public-share folder, I think this is not strictly necessary (I personally do use this as a way to share notes across my local devices but most users probably wouldn't need this).
Although, I should probably add xdg-desktop in the permission list as it's very common for users to save their notes in their desktop folder.

Reply:
so by design you don't use the FileChooser portal. You should.

Reply:
Yes, that's (somewhat) correct. Scrivano doesn't use native file dialogs (although this may be implemented for *some* things, e.g. importing pictures, in the future). As I mentioned, the choice isn't just by design but dictated by some features that are needed by the app (for example, as I mentioned above, the ability to save files using an interval based timer which needs to happen without user intervention and this, as far as I know, wouldn't be possible using the portal api).